numpy/lib/tests/test_io.py

```diff
@@ -2728,4 +2728,7 @@ def test_load_refcount():
     dt = [("a", 'u1', 2), ("b", 'u1', 2)]
     with assert_no_gc_cycles():
         x = np.loadtxt(TextIO("0 1 2 3"), dtype=dt)
-        assert_equal(x, np.array([((0, 1), (2, 3))], dtype=dt))
+        assert_equal(x, np.array(((0, 1), (2, 3)), dtype=dt))
+        # assert_equal uses a recursive function which
+        # makes this context unhappy.
+        break_cycles()
```

Comment:
Can you reorganize it to move the `assert_equal` out of the `with` statement?  What we care about is loadtxt, not the assertion function.

Reply:
Done.